scalableminds/webknossos
# Combine custom names of merged agglomerates & transfer custom name to split-off segments

This PR improves the way custom names of segments / agglomerates are handled during proofreading. Previously, the potential custom name of the target segment was dropped in a merge action and a split-off segment always "lost" its previous custom name. Now, the custom names are merged in case of a merge and split-off segments inherit the custom name of the original agglomerate.

### URL of deployed dev instance (used for testing):
- https://___.webknossos.xyz

### Steps to test:
- Open a dataset with a segmentation layer with a hdf5 mapping.
- load 2 segments & give each a custom name
- merge these segments. The newly merged segment should have both custom names separated by a comma.
- Merge two segments out of which only one has a custom name. The merged segment should have retained the custom name but not the default one.
- Split of a segment from an agglomerate with a custom name. The split-off segment should have the same name as the original agglomerate.   
- Split a segment from all its neighbors. All new split off neighbors should have the custom name of the original segment in case the original segment had a custom name.

### TODOs:
- [x] Check whether there are other proofreading actions that should be considered.

### Issues:
- fixes [#7830](https://github.com/scalableminds/webknossos/issues/7830)

------
(Please delete unneeded items, merge only when none are left open)
